src/frontend/src/lib/stores/modal.store.ts

```diff
@@ -32,20 +32,21 @@ export interface Modal<T> {
 		| 'btc-transaction'
 		| 'dapp-details';
 	data?: T;
+	modalId?: symbol;
 }
 
 export type ModalData<T> = Option<Modal<T>>;
 
 export interface ModalStore<T> extends Readable<ModalData<T>> {
-	openEthReceive: <D extends T>(data: D) => void;
-	openIcpReceive: <D extends T>(data: D) => void;
-	openIcrcReceive: <D extends T>(data: D) => void;
-	openCkBTCReceive: <D extends T>(data: D) => void;
-	openCkETHReceive: <D extends T>(data: D) => void;
-	openBtcReceive: <D extends T>(data: D) => void;
-	openReceive: <D extends T>(data: D) => void;
-	openSend: <D extends T>(data: D) => void;
-	openBuy: <D extends T>(data: D) => void;
+	openEthReceive: (modalId: symbol) => void;
+	openIcpReceive: (modalId: symbol) => void;
+	openIcrcReceive: (modalId: symbol) => void;
+	openCkBTCReceive: (modalId: symbol) => void;
+	openCkETHReceive: (modalId: symbol) => void;
+	openBtcReceive: (modalId: symbol) => void;
+	openReceive: (modalId: symbol) => void;
+	openSend: (modalId: symbol) => void;
```

Comment:
This one `modalId` is not set (see `Send` cmp). We improve that in another PR?

Reply:
It is actually used in the sub-component `SendButtonWithModal`. I think the idea is that this specific button is used only once, while the `Receive` ones is used both for receive and convert (with different props), so they need a `modalId` one level higher